### ISSUE TITLE
`boolean-prop-naming` triggers "Cannot read property 'type' of undefined"

### ISSUE THREAD

**Issue by mindtricksdev** [yannickcr/eslint-plugin-react#1369]
Hello,

When I add this rule to my .eslintrc file Visual Studio Code triggers a popup error `Cannot read property 'type' of undefined`.

This is my rule:
`"react/boolean-prop-naming": [1, { "rule": "(^(is|has)[A-Z]([A-Za-z0-9]?)+)|(^[a-z]+(ed|able)$)" }],`

I think this is related somehow to the spread operator as the errors trigger when I type `...test`

**Comment by yannickcr** [comment]
Can you show us the full piece of code that trigger the crash ? It will help us to fix the issue.

**Comment by mindtricksdev** [comment]
Here's a screenshot:

![image](https://user-images.githubusercontent.com/5068284/29286105-76f184d8-8139-11e7-8abb-9467c45dac2b.png)


The error is triggered after typing `...x`.

Not sure why, but if I disable the boolean rule the error stops occurring.

**Comment by EvHaus** [comment]
I can reproduce this. Will fix.

**Comment by setek** [comment]
Hi, I’m running 7.4.0 and still getting this error:

**package.json:**
`"eslint-plugin-react": "^7.4.0"`

`eslint` is 4.7.2

**.eslintrc.js:**
```
"react/boolean-prop-naming": [                               // Enforce consistent naming of boolean props with is or has
    "error",
    { "rule": "^(is|has)[A-Z]([A-Za-z0-9]?)+" }
],
```

**Error:**
```
Cannot read property 'properties' of undefined
TypeError: Cannot read property 'properties' of undefined
    at EventEmitter.MemberExpression (/Users/user/repos/myrepo/node_modules/eslint-plugin-react/lib/rules/boolean-prop-naming.js:183:45)
    at emitOne (events.js:120:20)
    at EventEmitter.emit (events.js:210:7)
    at NodeEventGenerator.applySelector (/Users/user/repos/myrepo/node_modules/eslint/lib/util/node-event-generator.js:265:26)
    at NodeEventGenerator.applySelectors (/Users/user/repos/myrepo/node_modules/eslint/lib/util/node-event-generator.js:294:22)
    at NodeEventGenerator.enterNode (/Users/user/repos/myrepo/node_modules/eslint/lib/util/node-event-generator.js:308:14)
    at CodePathAnalyzer.enterNode (/Users/user/repos/myrepo/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:606:23)
    at Traverser.enter (/Users/user/repos/myrepo/node_modules/eslint/lib/linter.js:962:32)
    at Traverser.__execute (/Users/user/repos/myrepo/node_modules/estraverse/estraverse.js:397:31)
    at Traverser.traverse (/Users/user/repos/myrepo/node_modules/estraverse/estraverse.js:501:28)
```

**Comment by ljharb** [comment]
@setek please file a new issue, and ideally include the code it's crashing on :-)
